src/backer/cardaning.py

```diff
@@ -69,61 +73,67 @@ def __init__(self, hab, ks=None):
+
+                # Metadata. accept int key type
+                builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
+                try:
+                    signed_tx = builder.build_and_sign([self.payment_signing_key],
+                                                    change_address=self.spending_addr,
+                                                    merge_change=True)
 
-                    if trans_size + kel_size >= MAX_TRANSACTION_SIZE:
+                    if len(signed_tx.to_cbor()) > MAX_TRANSACTION_SIZE - MAX_TRANSACTION_SIZE_MARGIN:
                         break
 
-                    # Build transaction
-                    builder = pycardano.TransactionBuilder(self.context)
-                    builder.add_input_address(self.spending_addr)
-                    builder.add_output(pycardano.TransactionOutput(self.spending_addr, pycardano.Value.from_primitive([TRANSACTION_AMOUNT])))
-
-                    submitting_kel.append(kel_item.decode('utf-8'))
-                    kel_data = kel_data + kel_item
-                    kel_data_bytes = bytes(kel_data)
-
-                    # Chunk size
-                    # bytearrays is not accept
-                    value = [kel_data_bytes[i:i + 64] for i in range(0, len(kel_data_bytes), 64)]
+                    temp_tx_cbor = signed_tx.to_cbor()
+                except pycardano.exception.InvalidTransactionException as e:
+                    if "exceeds the max limit" in str(e):
+                        break
 
-                    # Metadata. accept int key type
-                    builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
-                    signed_tx = builder.build_and_sign([self.payment_signing_key],
-                                                    change_address=self.spending_addr,
-                                                    merge_change=True)
-                    tx_cbor = signed_tx.to_cbor()
-                    trans_size = len(tx_cbor)
-                    self.pending_kel.remove(kel_item)
-
-                if tx_cbor:
-                    # Submit transaction
-                    self.context.submit_tx(tx_cbor)
-                    transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
-                    trans = {
-                        "id": transId,
-                        "kel": submitting_kel,
-                        "tip": self.tipHeight
-                    }
-                    self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
-        except Exception as e:
-            logger.critical(f"Cannot submit transaction: {e}")
+                submitting_tx_cbor = bytes(temp_tx_cbor)
+                submitting_kel.append(event.decode('utf-8'))
+                self.keldb_queued.rem(keys=keys)
+                self.keldb_published.pin(keys=keys, val=serder)
```

Comment:
- I would add to published before removing from queued in case we crash in between the calls and lose the data
- Can we just bulk store them after submitting the tx, in case e.g. the node is down?

Reply:
Yes, It's better to process that (add to publish then remove from queued) after submitting the tx

Reply:
fixed